app/user/rest.py

```diff
@@ -142,41 +136,75 @@ def verify_user_code(user_id):
         increment_failed_login_count(user_to_verify)
         raise InvalidRequest("Code has expired", status_code=400)
 
-    if code_type == 'sms':
-        user_to_verify.current_session_id = str(uuid.uuid4())
-        user_to_verify.logged_in_at = datetime.utcnow()
-        user_to_verify.failed_login_count = 0
-        save_model_user(user_to_verify)
+    user_to_verify.current_session_id = str(uuid.uuid4())
+    user_to_verify.logged_in_at = datetime.utcnow()
+    user_to_verify.failed_login_count = 0
+    save_model_user(user_to_verify)
 
     use_user_code(code.id)
     return jsonify({}), 204
 
 
 @user_blueprint.route('/<uuid:user_id>/sms-code', methods=['POST'])
 def send_user_sms_code(user_id):
+    data = request.get_json()
+    user_to_send_to = validate_2fa_call(user_id, data, post_send_user_sms_code_schema)
+    if not user_to_send_to:
+        return jsonify({}), 204
```

Comment:
If this fails shouldn't you return an error?

Reply:
hmm, i was just following existing behaviour that threw away the errors object (`verify_code, errors = request_verify_code_schema.load(request.get_json())`) - i imagine that was a dev mistake rather than intentional. will look into it and see if any tests break

Reply:
it's deliberate - see `test_send_sms_code_returns_204_when_too_many_codes_already_created`